src/UCTSearch.cpp

```diff
@@ -405,10 +405,15 @@ Move UCTSearch::think(BoardHistory&& new_bh) {
 #endif
 
     // set up timing info
-
     Time.init(bh_.cur().side_to_move(), bh_.cur().game_ply());
-    m_target_time = (Limits.movetime ? Limits.movetime : Time.optimum()) - cfg_lagbuffer_ms;
     m_max_time    = Time.maximum() - cfg_lagbuffer_ms;
+    
+    if (Limits.movetime) {
+        m_target_time = Limits.movetime;
```

Comment:
This should have the lagbuffer subtracted.
(

Reply:
Limits.movetime did not have it subtracted before, so I didn't change that at all

Reply:
the previous code was m_target_time = (move time conditional dependent expression) - lagbuffer.
So lag buffer was applied regardless of movetime or not.